xarray/coding/cftimeindex.py

```diff
@@ -541,14 +541,29 @@ def shift(self, n, freq):
         >>> index.shift(1, "M")
         CFTimeIndex([2000-02-29 00:00:00],
                     dtype='object', length=1, calendar='gregorian', freq=None)
+        >>> index.shift(1.5, "D")
+        CFTimeIndex([2000-02-01 12:00:00],
+                    dtype='object', length=1, calendar='gregorian', freq=None)
         """
         from .cftime_offsets import to_offset
 
-        if not isinstance(n, int):
-            raise TypeError(f"'n' must be an int, got {n}.")
+        if not isinstance(n, (int, float)):
+            raise TypeError(f"'n' must be an int or float, got {n}.")
         if isinstance(freq, timedelta):
             return self + n * freq
         elif isinstance(freq, str):
+            if "us" in freq and not isinstance(n, int):
```

Comment:
```suggestion
            if "us" in freq and isinstance(n, float):
```
This limitation also isn't documented in the docstring. What happens with "ns"?